
package PDcodec;

message LogTxnOp {
	required uint64 txn_id = 1;
}

message LogData {
	required string table = 1;
	required uint64 txn_id = 2;
	required uint32 recmask = 3;
	required bytes key = 4;
	optional bytes value = 5;
}

message LogTable {
	required string tabname = 1;
	required uint64 txn_id = 2;
	required uint32 recmask = 3;
	required uint64 root_id = 4;
}

message RootEnt {
	required bytes key = 1;
	required uint64 file_id = 2;
}

message RootIdx {
	repeated RootEnt entries = 1;
}

message TableMeta {
	required string name = 1;
	required string uuid = 2;
	required uint64 root_id = 3;
}

message Superblock {
	required string uuid = 1;
	required uint64 log_id = 2;
	required uint64 next_txn_id = 3;
	required uint64 next_file_id = 4;
	repeated TableMeta tables = 5;
}

